### PR TITLE
Implements slack client actor

### PR DESCRIPTION
**Pull request by synk**
I believe `SlackClient` should be an actor in order to deal with api limit.

### CODE REVIEW

**Comment by sifue** [comment]
Thank you for your pull request.
It is looking good to change `SlackClient` from a singleton to  an actor. 
I will merge this after checking this. Thanks.

File: src/main/scala/org/soichiro/ircslackrelay/SlackClientActor.scala [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  private def idling(): Receive = {
+    case PostToSlack(message, channelName) =>
+      slackClient.postMessage(message, channelName, log)
+      context.become(running())
+      scheduleAwake()
+  }
+
+  private def running(queue: Queue[PostToSlack] = Queue.empty): Receive = {
+    case post@PostToSlack(message, channelName) =>
```
Comment (sifue) [inline src/main/scala/org/soichiro/ircslackrelay/SlackClientActor.scala]: message and channelName were not used.

**Comment by sifue** [comment]
I tested and results is ok.
After merged, I will fix a few.

**Comment by synk** [comment]
Thank you for the merge! :bow:

**Comment by sifue** [comment]
I tagged 1.3.1 and refactor `SlackClientActor` to FSM. 
https://github.com/sifue/ircslackrelay/commit/0a5b16373679af87df621a09b8dbefb7eeb4a8b2
Thanks for your contribution.

**Comment by synk** [comment]
Nice fix :+1:

**Pull request merged**